File: code/simian/simian-master-v1.6/SimianJS/Tests/test.Array.lua
#!/usr/bin/env luajit
local count = 10000000
local active = count/2
local clique = 20
local random = math.random

local start = os.clock()
local A = {}
for i=1,count do A[i] = random() end

local value = 0.0
local opCount = 0
for i=1,active do
    for j=1,clique do
        value = value + A[i]*random()
        opCount = opCount + 1
    end
end

local finish = os.clock() - start

print(value, opCount)
print("Took: " .. finish .. "(s)")

File: code/simian/simian-master-v1.6/SimianJS/Tests/test.Array.js
"use strict";
"use wasm";

var print = print;
if (!print) print = masala.io.print;

const count = 10000000, active = count/2, clique = 20, random = Math.random;

var start = new Date();
var A = new Array(count);
for (var i = 0; i < active; i++) A[i] = random();

var value = 0.0;
var opCount = 0;
for (var i = 0; i < active; i++)
    for (var j = 0; j < clique; j++) {
        value += A[i]*random();
        opCount++;
    }

var finish = new Date() - start;

print(value, opCount);
print("Took: " + (finish/1000) + "(s)");
